### ISSUE TITLE
Negative Julian Dates?

### ISSUE THREAD

**Issue by aaschwanden** [Unidata/netcdf4-python#584]
Hi,

with our ice sheet model we frequently perform simulations spanning full glacial cycles. Our time axis is compatible with CF Conventions:

```
        double time(time) ;
                time:units = "seconds since 1-1-1" ;
                time:axis = "T" ;
                time:calendar = "365_day" ;
                time:long_name = "time" ;
                time:bounds = "time_bounds" ;

```

I realize that netcdftime cannot handle negative Julian dates, as I'm getting:
```
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/netcdftime/netcdftime.py", line 396, in _DateFromNoLeapDay
    raise ValueError('Julian Day must be positive')
ValueError: Julian Day must be positive
```
This causes trouble, as most of my postprocessing tools rely on netcdf (my own script, OCGIS, TimeManager plugin in QGIS).

Technically, in postprocessing, I could create a 'fake' time axis, do the postprocessing, and then revert to the original time axis. However, due to the file sizes (can be several 100 GB per file), this is unfeasible.

Maybe there is another workaround? Any suggestions are greatly appreciated.

**Comment by jswhit** [comment]
No workaround that I know offhand - I don't the the algorithms I use can handle negative Julian days (and the Julian calendar is not valid in that case anyway).  There must be a way to deal with this though, but I haven't ever given it much thought.

**Comment by aaschwanden** [comment]
Thanks.

Maybe one could handle the '360_day', '365_day', and '366_day' calendars differently, since here the concept of Julian days does not apply anyway?

To have such capabilities would be very useful (I assume for anyone doing paleo simulations). Maybe my colleague @ckhroulev has an idea.

**Comment by jswhit** [comment]
I use the Meeus algorithm (from Meeus, Jean (1998) Astronomical Algorithms (2nd Edition). Willmann-Bell, Virginia. p. 63), which is restricted to positive julian day numbers (back to 4800 BC). 

There is an algorithm here

http://aa.quae.nl/en/reken/juliaansedag.html

that supposedly works for negative Julian day numbers.

**Comment by ckhroulev** [comment]
@jswhit Thanks for the link! I'll give it a try and let you (and @aaschwanden) know how it goes.

**Comment by jswhit** [comment]
merged pull request #586, which allows for negative julian days in 360_day and 365_day calendars.
